src/legacy/core_plugins/visualizations/public/np_ready/public/wizard/show_new_vis.tsx

```diff
@@ -23,7 +23,7 @@ import ReactDOM from 'react-dom';
 import { I18nProvider } from '@kbn/i18n/react';
 import { IUiSettingsClient, SavedObjectsStart } from 'kibana/public';
 import { NewVisModal } from './new_vis_modal';
-import { TypesStart } from '../../../../../visualizations/public/np_ready/public/types';
+import { TypesStart } from '../types';
```

Comment:
currenty `NewVisModal` component and `showNewVisModal` function are statically exported and receive all dependencies as props/parameters.

i think it makes sense to rather export them on the contract so most of the parameters can be provided and plugin setup. calling showNewVisModal should be as simple as:

```
npStart.plugins.visualizations.showNewVisModal();
```
```

Reply:
I ended up with:

1. expose showNewVisModal() on vis plugin - as you suggested. But in vis embeddable factory I am still using it as function - https://github.com/elastic/kibana/pull/56654/files#diff-a45935fc131c617ca6ad658838e474ebR40. As it is part of the same plugin. Not sure if this is correct? 
 
2. I started using it in visualise_listing and removed custom angular directive wrapper around NewVisModal react component